test-suite/testkit/src/builder.rs

```diff
@@ -172,13 +178,20 @@ impl TestKitBuilder {
     }
 
     /// Adds a Rust service to the testkit.
-    pub fn with_rust_service(mut self, service: impl Into<InstanceCollection>) -> Self {
-        let InstanceCollection { factory, instances } = service.into();
-        self.rust_runtime = self.rust_runtime.with_available_service(factory);
-        self.instances.extend(instances);
+    pub fn with_rust_service(mut self, service: impl Into<Box<dyn ServiceFactory>>) -> Self {
+        let service = service.into();
+        self.rust_runtime = self.rust_runtime.with_factory(service);
         self
     }
 
+    /// Adds a Rust service that has default instance configuration to the testkit. Corresponding
+    /// artifact and default instance are added implicitly.
+    pub fn with_rust_service_default(self, service: impl DefaultInstance) -> Self {
```

Comment:
Nit: `with_default_rust_service` looks like a better name.

Reply:
I'd suggest to use `builtin` instead of `default`.

Reply:
I feel like `with_builtin_rust_service` causes `DefaultInstance` -> `BuiltinInstance` (as @popzxc  suggested). On the one hand, `Default...` communicates that it provides default values for a service. But on the other hand, the actual implementation is about builtin services and probably the meaning of "builtin service" is more close to blockchain in this case than "default service". So, `builtin...` sounds better.